modules/account-lib/test/unit/coin/dot/transactionBuilder/base.ts

```diff
@@ -178,4 +181,183 @@ xdescribe('Dot Transfer Builder', () => {
       should.deepEqual(builder.getTip(), undefined);
     });
   });
+
+  describe('add signature', () => {
+    const factory = register('tdot', TransactionBuilderFactory);
+
+    it('should add a signature to transaction', async () => {
+      const transferBuilder = factory
+        .getTransferBuilder()
+        .amount('90034235235322')
+        .sender({ address: sender.address })
+        .to({ address: receiver.address })
+        .validity({ firstValid: 3933, maxDuration: 64 })
+        .referenceBlock('0x149799bc9602cb5cf201f3425fb8d253b2d4e61fc119dcab3249f307f594754d')
+        .sequenceId({ name: 'Nonce', keyword: 'nonce', value: 200 })
+        .fee({ amount: 0, type: 'tip' });
+
+      transferBuilder.sign({ key: accounts.account1.secretKey });
+      const signedTx = await transferBuilder.build();
+      const signature = signedTx.signature[0];
+
+      // verify rebuilt transaction contains signature
+      const rawTransaction = signedTx.toBroadcastFormat() as string;
+      const rebuiltSignedTransaction = await factory
+        .from(rawTransaction)
+        .validity({ firstValid: 3933, maxDuration: 64 })
+        .referenceBlock('0x149799bc9602cb5cf201f3425fb8d253b2d4e61fc119dcab3249f307f594754d')
+        .build();
+      rebuiltSignedTransaction.signature.should.deepEqual(signedTx.signature);
+
+      const transferBuilder2 = factory
+        .getTransferBuilder()
+        .amount('90034235235322')
+        .sender({ address: sender.address })
+        .to({ address: receiver.address })
+        .validity({ firstValid: 3933, maxDuration: 64 })
+        .referenceBlock('0x149799bc9602cb5cf201f3425fb8d253b2d4e61fc119dcab3249f307f594754d')
+        .sequenceId({ name: 'Nonce', keyword: 'nonce', value: 200 })
+        .fee({ amount: 0, type: 'tip' });
+      transferBuilder2.addSignature({ pub: accounts.account1.publicKey }, Buffer.from(signature, 'hex'));
```

Comment:
Since the signature is an attribute used to construct the transaction, there is no need for the `add` prefix, you can simply to `builder.signature(...)`

Reply:
I can do this but sol is using `addSignature` and would prefer to stay consistent. 

I can update this for sol/dot in a follow up PR.

Reply:
@joey-bai - heads up, im gonna be making this change in a follow up

Reply:
@argjv @lcovar i wanted this to be `addSignature` because Solana supports multiple signatures. so rather than setting a single `signature` field, we're _adding_ signatures with each call to `addSignature`.

this is needed when creating staking accounts and durable nonce accounts.

Reply:
but i suppose, Solana's implementation can treat this as appending signatures.
other coins can treat this as a single signature.

Reply:
Sounds like we are talking about the same. Instead of:

```
transferBuilder2.addSignature({ pub: accounts.account1.publicKey }, Buffer.from(signature1, 'hex'));
transferBuilder2.addSignature({ pub: accounts.account2.publicKey }, Buffer.from(signature2, 'hex'));
transferBuilder2.addSignature({ pub: accounts.account3.publicKey }, Buffer.from(signature3, 'hex'));
```

You would have:

```
transferBuilder2.signature({ pub: accounts.account1.publicKey }, Buffer.from(signature1, 'hex'));
transferBuilder2.signature({ pub: accounts.account2.publicKey }, Buffer.from(signature2, 'hex'));
transferBuilder2.signature({ pub: accounts.account3.publicKey }, Buffer.from(signature3, 'hex'));
```

My change request for the name is because `addSignature` is a verb but what you are doing is setting a field just like `amount`, `sender`, `fee`, etc, so `signature` would be more proper naming. Otherwise, we would also do `addAmount`, `addSender`, `addFee`

Reply:
^ going to have a refactor PR open by EOW.